x/stake/genesis.go

```diff
@@ -29,6 +29,8 @@ func InitGenesis(ctx sdk.Context, keeper Keeper, data types.GenesisState) error
 
 		// Manually set indexes for the first time
 		keeper.SetValidatorByPubKeyIndex(ctx, validator)
+
+		validator.BondIntraTxCounter = int16(i) // set the intra-tx counter to the order the validators are presented
```

Comment:
shouldn't we set this value right away so its consistent ?

Reply:
Only the power store key uses this.

Reply:
@ebuchman as in set it in the genesis file as opposed to in this ordering? Yeah could be done, might be better to keep it more explicit. But then again. who get's to decide the order! I almost think for that organizational reason it should be set where it is currently.

Reply:
It's still effectively set by the genesis file though, since we iterate through the validator array.

Reply:
yeah I guess it's alphabetical - honestly I don't think this is a real concern it's an extreme edge case, we just need something in there for key collision